src/Microsoft.OData.Core/UriParser/TreeNodeKinds/QueryNodeKind.cs

```diff
@@ -158,6 +158,11 @@ public enum QueryNodeKind
         /// Count of a collection contains primitive or enum or complex or entity type.
         /// </summary>
         Count = InternalQueryNodeKind.Count,
+
+        /// <summary>
+        /// Cast on a single property.
+        /// </summary>
+        SinglePropertyCast = InternalQueryNodeKind.SinglePropertyCast,

```

Comment:
Rename to SingleValueCast as per discussion.

---
In reply to: [125102394](https://github.com/OData/odata.net/pull/848#discussion_r125102394) [](ancestors = 125102394,125081678,124892629)